source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -463,27 +463,44 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
 
-  // To enumerate the required builtin ciphers, curves, algorithms, and
-  // versions, uncomment '#define LOG_BUILTIN_STAT_NAMES' below, and run
-  //  bazel test //test/extensions/transport_sockets/tls/... --test_output=streamed
-  //      | grep " Builtin ssl." | sort | uniq
-  // #define LOG_BUILTIN_STAT_NAMES
-  //
-  // TODO(#8035): improve tooling to find any other built-ins needed to avoid
-  // contention.
-
-  // Ciphers
-  stat_name_set_->rememberBuiltin("AEAD-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-ECDSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-SHA");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-CHACHA20-POLY1305");
+  // Ciphers are configured as a string delimited by ":", with equivalence
+  // groups given as "[alt1|alt2]", and exclusions preceded by "!". For the
+  // purposes of collecting stats, we want to track all these names. We don't
+  // need to fully parse out the structure of the cipher suites. Just extract
+  // out the names. We skip exclusions as those do not show up as entire
+  // stat-name segments, so they would not match.
+  for (absl::string_view cipher_suite :
+       absl::StrSplit(config.cipherSuites(), absl::ByAnyChar(":|[]"))) {
+    if (!cipher_suite.empty() && cipher_suite[0] != '!') { // skip exclusions and empty strings.
+      stat_name_set_->rememberBuiltin(cipher_suite);
+    }
+  }
+
+  // This cipher is referenced from
+  // IpVersionsClientVersions/SslCertficateIntegrationTest.ServerRsa/IPv4_TLSv1_3,
+  // possibly due to the call to
+  // ClientSslTransportOptions().setSigningAlgorithmsForTest
+  // in test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc, function
+  // rsaOnlyClientOptions.
   stat_name_set_->rememberBuiltin("TLS_AES_128_GCM_SHA256");
 
+  // This cipher's appearance appears to be induced by setting "cipher_suites" to
+  // "TLS_RSA_WITH_AES_128_GCM_SHA256" in the configuration for
+  // SslSocketTest.RsaPrivateKeyProviderAsyncDecryptSuccess in
+  // test/extensions/transport_sockets/tls/ssl_socket_test.cc
+  stat_name_set_->rememberBuiltin("AES128-GCM-SHA256");
```

Comment:
This should be covered by the "configured cipher suites" alternative suggested above.